test/topology_custom/test_mv_delete_partitions.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, key, values, concurrency):
+    logger.info(f"Starting writes for key {key} with concurrency {concurrency}")
+    async def do_inserts(i: int):
+        # Insert all rows with keys with the same modulo as i
+        insert_stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+        while i < values:
+            await cql.run_async(insert_stmt, [key, i, key + i])
+            # await cql.run_async(f"INSERT INTO ks.tab (key, c, v) VALUES ({key}, {i}, {key + i})")
+            i += concurrency
+    tasks = [asyncio.create_task(do_inserts(i)) for i in range(concurrency)]
+
+    await asyncio.gather(*tasks)
+    logger.info(f"Finished writes for key {key} with concurrency {concurrency}")
+
+async def wait_for_views(cql, mvs_count, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        logger.info(f"Views built: {done[0][0]}")
+        if done[0][0] == node_count * mvs_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+# This test reproduces issue #12379
+# The test scenario is the following:
+# 1. Create a table with 30000 rows in 30 partitions (1000 rows per partition)
+# 3. Create and wait for 30 MVs
+# 4. Delete all partitions, causing 30*30*1000 = 0.9M view updates
+# If the test doesn't cause a crash due to OOM in the step 4, it passes
+@pytest.mark.asyncio
+@skip_mode('debug', "-m option doesn't work in debug mode")
```

Comment:
@avikivity I wonder if we could add support for -m option in debug mode and if it's worth the hassle. IIRC in debug mode we use the system malloc(), so it's harder to limit the memory use, but it's still possible to limit it either in code (just have an malloc/free wrapper that counts the allocation) or perhaps more easily using "ulimit".